### ISSUE TITLE
Please make a release? (+due questions re: maintenance)

### ISSUE THREAD

**Issue by aldanor** [hyperopt/hyperopt#348]
Dear authors/maintainers,

Would you please make a tarball release so that 

1) Users installing it via pip install don't run into networkx 2.0 incompatibility which is fixed on master
2) We can build a conda-forge hyperopt recipe (https://github.com/conda-forge/staged-recipes/pull/4710) so that the latest hyperopt is available via conda install

On a side note, as I've tried running the full test suite on all platforms while building the conda-forge package, more than a dozen different tests were failing on master. It looks like there's a rather significant amount of work to be done e.g. to catch up with the latest numpy, plus fixing the flaky tests. Is anyone planing to take on that? I could probably find time to help out, but it would then involve *major* cleanup and refactoring so the codebase/tests are easier/more pleasant to work with from development standpoint which I'm not sure the authors are okay with. Ideally, there should also automated tests, i.e. tox/travis setup, which would also simplify the process of conda releases and reduce downstream surprises.

// @jaberg? @dwf?

Thanks a million!

**Comment by jaberg** [comment]
Hi Ivan, thanks for bringing this up.

 I will spend some time on hyperopt in the next couple of weeks. I can set
up e.g. travis and fix the tests. At that point a new pypi version would
make sense. Could I ask you for code review on those changes?

You are welcome to do the conda recipe, keep the list posted.

For the code refactoring and/or cleanup that could be great, lets have a
chance to discuss on list or github issues before anyone does a lot of work
to make sure the whole process to merge goes smoothly. Where would you like
to start?

On Dec 27, 2017 4:31 AM, "Ivan Smirnov" <notifications@github.com> wrote:

Dear authors/maintainers,

Would you please make a tarball release so that

   1. Users installing it via pip install don't run into networkx 2.0
   incompatibility which is fixed on master
   2. We can build a conda-forge hyperopt recipe (
   conda-forge/staged-recipes#4710
   <https://github.com/conda-forge/staged-recipes/pull/4710>) so that the
   latest hyperopt is available via conda install

On a side note, as I've tried running the full test suite on all platforms
while building the conda-forge package, more than a dozen different tests
were failing on master. It looks like there's a rather significant amount
of work to be done e.g. to catch up with the latest numpy, plus fixing the
flaky tests. Is anyone planing to take on that? I could probably find time
to help out, but it would then involve *major* cleanup and refactoring so
the codebase/tests are easier/more pleasant to work with from development
standpoint which I'm not sure the authors are okay with. Ideally, there
should also automated tests, i.e. tox/travis setup, which would also
simplify the process of conda releases and reduce downstream surprises.

// @jaberg <https://github.com/jaberg>? @dwf <https://github.com/dwf>?

Thanks a million!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/hyperopt/hyperopt/issues/348>, or mute the thread

**Comment by aldanor** [comment]
@jaberg Hi James, thank you for your reply. I'd be glad to do a code review, of course, or discuss how to fix the current test failures and/or help with setting up automated testing.

As for the refactoring/cleanup -- I have quite a few ideas on how to clean things up in a major way to hopefully simplify future maintenance. I could try to sum it up in a separate GH issue in the coming days, if you would.

**Comment by jaberg** [comment]
Sounds good, I will look out for those GH issues. If you remember, post
links to those issues to this thread too.

**Comment by jaberg** [comment]
Original question was to do a tarball release. What do you mean exactly? What would it take to make this done?

**Comment by nouiz** [comment]
github allow to get a tarball of the dev version. One example:

https://github.com/hyperopt/hyperopt/archive/master.zip

**Comment by ClimbsRocks** [comment]
Really glad to see this project coming back to life. 

It's impressive how quickly dependency packages like numpy are updating, but also a bummer to see how frequently those changes are breaking. The last PyPi release for this package was just over a year ago, and yet it doesn't work any longer, due to how rapidly our space is moving. 

I've found it enormously helpful to pin supported version ranges for [auto_ml](https://github.com/ClimbsRocks/auto_ml). If someone releases a breaking change to a package we rely on (seems to happen every other month or so, with Keras being the most recent culprit), we just [specify in our setup.py](https://github.com/ClimbsRocks/auto_ml/blob/master/setup.py#L67) that we don't support that version yet. 

It's pretty trivial from there to cut a new release and push it to PyPi, and work on support later when we've got more time for it (or let a contributor take it on who wants some cool functionality that's baked into the new Keras release!). 

Really glad to see a community building this project back up. Can't wait to put the new version to use.
